worker/src/RTC/RtpPacket.cpp

```diff
@@ -689,12 +713,22 @@ namespace RTC
 			ptr += size_t{ this->payloadPadding };
 		}
 
-		MS_ASSERT(static_cast<size_t>(ptr - buffer) == this->size, "ptr - buffer == this->size");
-
 		// Create the new RtpPacket instance and return it.
-		auto* packet = new RtpPacket(
+		auto* packet = RtpPacketPool.Allocate();
+		new (packet) RtpPacket(
 		  newHeader, newHeaderExtension, newPayload, this->payloadLength, this->payloadPadding, this->size);
 
+		SharedPtr shared(
+		  packet,
+		  /*Deleter*/
+		  [](RtpPacket* packet)
```

Comment:
@nazar-pc could it be possibly to put this lambda as deleter struct defined once?
```
struct RtpPacketPoolDeleter
{
    void operator()(RtpPacket* packet) const { 
        // Call destructor manually since memory was pre-allocated upfront.
        packet->~RtpPacket();
        // Return packet into object pool for future reuse of memory allocation.
        RtpPacketPool.Return(packet);
    }`
};
```
And then use as `std::shared_ptr<RtpPacket> packet(new RtpPacket(), RtpPacketPoolDeleter());`

Reply:
I think so, but I'll leave that for the future. There are still other places that could use shared_ptr with custom deleter that are not refactored yet. Feel free to create a PR if you feel like it.